examples/style.rs

```diff
@@ -0,0 +1,238 @@
+                            4,
+                            1,
+                        ),
+                    );
+                }
+            }
+        }
+    }
+}
+
+fn render_indexed_grayscale<B: Backend>(frame: &mut Frame<B>) {
+    // Dark grayscale ramp
+    for j in 0..6 {
+        for k in 0..2 {
+            let color = Color::Indexed(j * 2 + k + 232);
+            let paragraph = Paragraph::new(color.to_string()).style(Style::default().bg(color));
+            frame.render_widget(paragraph, Rect::new(k as u16 * 6 + 52, j as u16 + 9, 6, 1));
+        }
+    }
+    // Light grayscale ramp
+    for j in 6..12 {
+        for k in 0..2 {
+            let color = Color::Indexed(j * 2 + k + 232);
+            let paragraph = Paragraph::new(color.to_string())
+                .style(Style::default().bg(color).fg(Color::Black));
+            frame.render_widget(paragraph, Rect::new(k as u16 * 6 + 52, j as u16 + 10, 6, 1));
+        }
+    }
+}
+
+fn render_modifiers<B: Backend>(frame: &mut Frame<B>) {
+    frame.render_widget(
+        Paragraph::new("Note: not all terminals support all modifiers")
+            .style(Style::default().fg(Color::Red).add_modifier(Modifier::BOLD)),
+        Rect::new(0, 30, 65, 1),
+    );
+    // Text Modifiers
+    for (j, bg, fg) in [
+        (0, Color::Black, Color::Gray),
+        (1, Color::Black, Color::DarkGray),
+        (2, Color::Black, Color::White),
+        (3, Color::DarkGray, Color::Black),
+        (4, Color::DarkGray, Color::Gray),
+        (5, Color::DarkGray, Color::White),
+        (6, Color::Gray, Color::Black),
+        (7, Color::Gray, Color::DarkGray),
+        (8, Color::Gray, Color::White),
+        (9, Color::White, Color::Black),
+        (10, Color::White, Color::DarkGray),
+        (11, Color::White, Color::Gray),
+    ] {
+        for (i, modifier) in once(Modifier::empty())
+            .chain(Modifier::all().iter())
+            .enumerate()
+        {
+            let paragraph = Paragraph::new(format!("{modifier:?}"))
+                .style(Style::default().fg(fg).bg(bg).add_modifier(modifier));
+            let y = i as u16 / 5 + 31 + j * 2;
+            // this method tends to overflow the terminal height if the terminal is too small
+            if y >= frame.size().height {
+                break;
+            }
+            frame.render_widget(paragraph, Rect::new((i as u16 * 13) % 65, y, 12, 1));
+        }
+    }
+}
+
+fn setup_terminal() -> Result<Terminal<CrosstermBackend<Stdout>>> {
+    enable_raw_mode()?;
+    let mut stdout = io::stdout();
+    execute!(stdout, EnterAlternateScreen)?;
+    let backend = CrosstermBackend::new(stdout);
+    let mut terminal = Terminal::new(backend)?;
+    terminal.hide_cursor()?;
+    Ok(terminal)
+}
+
+fn restore_terminal(mut terminal: Terminal<CrosstermBackend<Stdout>>) -> Result<()> {
+    disable_raw_mode()?;
+    execute!(terminal.backend_mut(), LeaveAlternateScreen)?;
+    terminal.clear()?;
```

Comment:
nitpick: I like not having `terminal.clear()` in restore, I feel like that is a better UX.

```suggestion
```

Reply:
Sounds good

Reply:
Done